### PR TITLE
[ONNX] Clean up diagnostic rules

### PR DESCRIPTION
**Pull request by BowenBao**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #107654
* __->__ #107653
* #107409
* #107408

Summary:

- Remove experimental rules that were never used.
- Fill in detailed rule descriptions.

### CODE REVIEW

File: torch/onnx/_internal/diagnostics/_rules.py
Comment (BowenBao) [inline torch/onnx/_internal/diagnostics/_rules.py:1]: This is codegen file, please ignore.

File: torch/csrc/onnx/diagnostics/generated/rules.h
Comment (BowenBao) [inline torch/csrc/onnx/diagnostics/generated/rules.h:1]: This is codegen file, please ignore.

**Review by justinchuby — APPROVED** [review]

File: torch/onnx/_internal/diagnostics/rules.yaml [hunk 1]
```diff
@@ -83,178 +79,98 @@
     deprecated: false
     tags: []
 
+## Rules for PyTorch (dynamo fx based) ONNX Exporter (FXE)
 
-
-- id: FXE0001
-  name: fx-tracer-success
+- id: FXE0007
```
Comment (titaiwangms) [inline torch/onnx/_internal/diagnostics/rules.yaml:84]: Should we re-start from one?

**Review by titaiwangms — APPROVED** [review]

File: torch/onnx/_internal/diagnostics/rules.yaml (hunk 1 above)
Comment (BowenBao) [reply torch/onnx/_internal/diagnostics/rules.yaml:84]: I'm leaving as is, technically it is taken and exposed in 2.0 ..

**Pull request closed without merging**